### ISSUE TITLE
JSON decoding problem

### ISSUE THREAD

**Issue by michalkkkd** [mangiucugna/json_repair#17]
**Describe the bug**
Failed JSON decoding

```
[6771] Failed to execute script 'jsonfix' due to unhandled exception!
Traceback (most recent call last):
File "jsonfix.py", line 281, in repair_json
File "json/__init__.py", line 348, in loads
File "json/decoder.py", line 337, in decode
File "json/decoder.py", line 353, in raw_decode
json.decoder.JSONDecodeError: Invalid control character at: line 2 column 48 (char 50)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "jsonfix.py", line 310, in <module>
File "jsonfix.py", line 283, in repair_json
File "jsonfix.py", line 17, in parse
File "jsonfix.py", line 29, in parse_json
File "jsonfix.py", line 106, in parse_object
File "jsonfix.py", line 59, in parse_json
ValueError: Invalid JSON format`
```

**To Reproduce**
Try to parse following json:

```
{
"real_content": "Some string: Some other string

Some string <a href=\"https://domain.com\">Some link</a>
"
}
```

**Expected behavior**
Correct json.

**Comment by mangiucugna** [comment]
Hi @michalkkkd I can't reproduce the issue.

When running the provided string like so:

`print(repair_json('{"real_content": "Some string: Some other string Some string <a href=\\"https://domain.com\\">Some link</a>"}'))`

It returns the correct json

When running it with the wrong escaping:

`print(repair_json('{"real_content": "Some string: Some other string Some string <a href=\"https://domain.com\">Some link</a>"}'))`

It returns a wrong dictionary but does not return a broken json.

I am running this on latest: `0.9.0`

**Comment by michalkkkd** [comment]
Hi @mangiucugna  :)
Github probably messed up when I pasted the snippet.
1. I was albo to fix it temporarily using this replacement in PHP  $data = str_replace('\"', '\'', $data);
2. I updated the library (sorry, for not doing it earlier). Now I get this result:
![image](https://github.com/mangiucugna/json_repair/assets/7553451/e4b907b3-657e-430f-abdd-eb068fcfb5ca)
So yes, json is not broken any more, but is unusable :(

**Comment by mangiucugna** [comment]
ok so this is an escaping issue and it's not trivial to address in general, I'll think about it but basically if you print that string, you can see that the escaping isn't passed to the function:
```
>>> print('{"real_content": "Some string: Some other string Some string <a href=\"https://domain.com\">Some link</a>"}')
{"real_content": "Some string: Some other string Some string <a href="https://domain.com">Some link</a>"}
```

Now, since it doesn't receive the escaping, there is no way for the library to handle the escaping and this is interpreted as a dictionary missing commas. The only (ugly) solution I can come up with is a regex for html tags that will escape those BUT it will break for cases in which the single comma is used like:

`repair_json('{"real_content": \'Some string: Some other string Some string <a href="https://domain.com">Some link</a>\'}')`

So need to give it more thoughts before I can deploy a fix for this type of breakage, a workaround is to use the following regex:

`re.sub(r'(?<!\\\\)"', r'\\"', input_string)`

That should fix the string (but again, I can't use that in json_repair because the escaping isn't passed over)

**Comment by michalkkkd** [comment]
I think, that your are extremely gifted to be able to create such an awesome string parsing library. Though I'm coding for years, low level string manipulation is an abstract concept for me.
Take your time!

**Comment by mangiucugna** [comment]
ok so I digged into this and the solution is still on your side, I will update the documentation for this problem but you should pass the string as a "raw string". Like:

json_string = r'{"real_content": "Some string: Some other string Some string <a href=\"https://domain.com\">Some link</a>"}'

This will pass the escaping correctly and fix your issue, however I noticed that if the string was broken (i.e. missing last brace) the library would get confused. So I pushed a fix for that in 0.10.1
